signer/src/signer/zone.c

```diff
@@ -58,7 +58,7 @@ zone_create(char* name, ldns_rr_class klass)
     if (!name || !klass) {
         return NULL;
     }
-    CHECKALLOC(zone = (zone_type*) malloc(sizeof(zone_type)));
+    CHECKALLOC(zone = (zone_type*) calloc(1, sizeof(zone_type)));
```

Comment:
Here you are really just masking a problem, namely that the zone_type isn't really fully initialized when it returns.  The initialization of ends should be moved/passed through this place.  It still remains uninitialized (but guaranteed to be NULL), but the code does not expect this entry not to be null....